src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/GenerateTypeDialogInProcess.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls.Primitives;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.Extensibility.Testing;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.Input;
+using Microsoft.VisualStudio.LanguageServices.Implementation.GenerateType;
+using Roslyn.Test.Utilities;
+using Roslyn.Utilities;
+
+namespace Roslyn.VisualStudio.NewIntegrationTests.InProcess
+{
+    [TestService]
+    internal partial class GenerateTypeDialogInProcess
+    {
+        private async Task<GenerateTypeDialog?> TryGetDialogAsync(CancellationToken cancellationToken)
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, cancellationToken);
+            return Application.Current.Windows.OfType<GenerateTypeDialog>().SingleOrDefault();
+        }
+
+        private async Task ClickAsync(Func<GenerateTypeDialog, ButtonBase> buttonAccessor, CancellationToken cancellationToken)
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            var dialog = await TryGetDialogAsync(cancellationToken);
+            AssertEx.NotNull(dialog);
+
+            Contract.ThrowIfFalse(await buttonAccessor(dialog).SimulateClickAsync(JoinableTaskFactory));
+        }
+
+        public async Task VerifyOpenAsync(CancellationToken cancellationToken)
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                if (await TryGetDialogAsync(cancellationToken) is null)
```

Comment:
Another place where we could use the helper (there's more further down as well, but I won't comment them all)

Reply:
Yeah, every dialog helper has the same code.